### ISSUE TITLE
Thumbnail is not changing from the WordPress media

### ISSUE THREAD

**Issue by pavanpatil1** [rtCamp/transcoder#253]
While changing the thumbnail from the media, it shows the thumbnail saved but while viewing the media the default thumbnail is visible. 

**Steps to reproduce:** 
1. Upload the media and edit it. 
2. Change the thumbnail video thumbnail and save. 
3. Observe the thumbnail is shown selected but while viewing the media the default thumbnail is visible. 

https://user-images.githubusercontent.com/44057535/205221160-977d123c-48bb-4fae-9cac-9bf0f48c2820.mp4

**Comment by javmah** [comment]
https://user-images.githubusercontent.com/21239064/205796752-dd79e0c4-aa80-4509-81e7-9bf058ecd70a.mp4

**Comment by javmah** [comment]
Dear Team members, 
Unfortunately, I am no longer in rtCamp, I fix this issue and created a [pull request](https://github.com/rtCamp/transcoder/pull/255). If you have any confusion or question regarding the pull request please contact me without any hesitation. I will be happy to help. thanks

**Comment by Pathan-Amaankhan** [comment]
Hello @gagan0123,
For this one, I think that we should re-analyze how things are stored in our DB.

The postmeta table stores the relative path of our images which should probably stored the IDs.
Also, we are not creating a new attachment for the images so, technically we won't be having any ids for them.

<img width="1552" alt="Screenshot 2023-02-02 at 2 41 54 PM" src="https://user-images.githubusercontent.com/63953699/216281831-fd03eb48-f1f3-4a86-bde0-2f9510ceca2d.png">

The approach which we can use is,
1. Create a taxonomy for attachments that will have `public` as false (Images in this taxonomy will be unlisted from the media library).
2. Add custom-generated video thumbnails under this taxonomy.
3. Save the ids of video thumbnails in `_rt_media_thumbnails`.
4. Now we can display these thumbnails in our edit section & update the `_thumbnail_id` as user selects the video thumbnail from custom-generated ones.

This way we won't be needing any additional `_rt_media_video_thumbnail` meta.
+
When the user selects the video thumbnail we can update image in featured image section + when the user selects the feature image from the media library we can uncheck all the radio buttons using js so that the user has an idea of what he/she has selected.

cc: @aviral-mittal

**Comment by dishitpala** [comment]
@Pathan-Amaankhan 

I don't think we should create a taxonomy for thumbnails because the relationship between thumbnails and video is one-to-one relation and ideally taxonomy is used when there is a one-to-many relation

If I am not wrong, thumbnails are not attachments here so we can not use them with id.

Can you add a featured image (An attachment in WordPress and not a thumbnail ) to the video and check if it appears as a video thumb in the frontend?

**Comment by Pathan-Amaankhan** [comment]
Hello @dishitpala,

I totally agree with you & have already tried adding a featured image. After adding the featured image to video everything goes fine but here is the catch:
- We want that if the user selects a thumbnail then it should be used as a featured image i.e. as [video poster](https://www.w3schools.com/tags/att_video_poster.asp).
- & as our thumbnails are not having any ids attached to them we cannot use them as featured images.
- So, if we want our thumbnails to act as a featured image, we have to add our thumbnails as WordPress attachments.
- This way we will be having ids through which we can add featured images but our gallery will be flooded with thumbnail images (As by default we are generating 5 thumbnails for each video).
- So, to prevent the overflooding of our Gallery, We can add a taxonomy & add all the thumbnails to that taxonomy then, we can hide images under that taxonomy from gallery.


So, here taxonomy that we will be adding will have no relationship with our video's attachments. It will be used just to hide our thumbnails in gallery.

cc: @joelabreo227 @gagan0123

**Comment by Pathan-Amaankhan** [comment]
<img width="1440" alt="Screenshot 2023-02-14 at 3 40 47 PM" src="https://user-images.githubusercontent.com/63953699/218705375-65924a5c-0782-4a34-b79d-dec2499641ba.png">

**Comment by dishitpala** [comment]
@Pathan-Amaankhan Earlier this functionality was working or from the beginning, it not working?

**Comment by Pathan-Amaankhan** [comment]
@dishitpala 

Yes, I tried this one on a dummy site & it is not working.

The above S.S. is from https://easydash.rtcamp.com/sites/details?siteName=amaan-transcoder.rt.gw.

Please have a look once.

**Comment by aviral-mittal** [comment]
@Souptik2001 Can you please look into this

**Comment by aviral-mittal** [comment]
@pavanpatil1 Did this issue get resolved once the server issue of transcoding was resolved?

**Comment by pavanpatil1** [comment]
@aviral-mittal, No this issue is still reproducible.
